content/en/smart-contracts/fundamentals/programmatic-storage/index.md

```diff
@@ -0,0 +1,60 @@
+1. **Data preparation**: Before storing files on Filecoin, clients must import them into their local Filecoin node, which produces a data CID (Content Identifier) - an ID uniquely describing the content. Then, the data is transferred to the storage provider.
+1. **Deal publication**: The deal is published on-chain, making the storage provider publicly accountable for the deal.
+1. Sealing and proving storage**: The storage provider stores the data in sectors. Sectors are the basic units of provable storage where storage providers store clients’ data and generate Proof-of-Spacetime (PoSt) on behalf of the Filecoin network. Sectors have standard sizes and a lifetime that storage providers can extend before reaching the end of the lifetime.
+
+## Programmatic storage
+
+The Filecoin virtual machine (FVM) is not designed to directly interact with data on Filecoin. Instead, it operates on blockchain state data, which includes metadata about stored data. The FVM's role is to compute over this state data and create logic around the storage and retrieval process. Storage and retrieval providers on the Filecoin network handle the actual process of storing and retrieving data.
+
+Develoeprs can use FVM actors (also called smart-contracts) to implement various use cases related to data storage and retrieval on the Filecoin network. For instance, they can automate the process of renewing storage deals, replicate data across multiple storage providers for redundancy, or implement access control rules for data retrieval.
+
+The FVM also includes an Ethereum Virtual Machine (EVM) runtime, enabling developers to deploy existing EVM-based smart contracts onto the FVM with little to no modifications. If you already have a smart contract written in Solidity, you can deploy it to Filecoin and have it interact with the Filecoin storage system.
+
+This process for programmatically storing data on Filecoin differs depending on how much data you need to store. Anything less than 4 GiB of data is considered _small_, while anything 4 GiB or above is considered _large_. Lets cover the process for both _small_ and _large_ storage deals:
+
+### Small-scale storage
+
+Storage deals for less than 4 GiB of data are considered _small_ on the Filecoin network. This is because storage providers use sectors of either 32 GiB or 64 GiB. Because of these comparably large sector sizes, SPs are incentivized to accept storage deals closer to this cap size. An SP will receive more FIL rewards for storing a 31 GiB file rather than a 200 KiB file. Clients will have difficulty finding a storage provider to accept deals for small amounts of data.
+
+To get around this problem, these developers can _aggregate_ their data into bundles. The deal-aggregation process bundles a collection of small storage deals with many others to create one large deal. 
```

Comment:
Can we end with "Read more about aggregation and how to build your aggregator here [hyperlink to the incoming Aggregation doc]"

Reply:
Since these pages are landing in the docs in separate PRs, we can't link out to future pages that don't yet exist without jeopardising the build process.. We can add in cross-page hyperlinks once everything's up.